### DISCUSSION TITLE
Clarification on functions as context

### DISCUSSION THREAD

**Discussion by thegrandpoobah** [mustache/spec#166] [Q&A] [answered]
According to the spec (from the section YAML, not mentioned in interpolation YAML):

```
5) Otherwise, the data is the value returned by calling the method with the given name.
```

So the following invocation:

```
Mustache.render('{{foo}}', { foo: function() { return null; } });
```

Would render out as (at least in JavaScript anyways) `null` the way I am interpreting the spec. But that seems a little on the sub-optimal side. Did I mis-interpret the sepc or is this another one of those language dependent traps?

Thanks for the clarification.

**Comment by jgonggrijp** [comment, marked as answer]
You interpreted the spec correctly, at least in the sense that the returned value is used for interpolation. That template, given those data, will render `` (the empty string) rather than `null`, because of this spec:

https://github.com/mustache/spec/blob/982aa9bf9d68b08f1baef049c289196e781ed2ab/specs/interpolation.yml#L104-L108

What do you mean by "sub-optimal"? What would be more optimal in your opinion?
